memory.c

```diff
@@ -91,6 +91,67 @@ read6502(uint16_t address)
 	}
 }
 
+/*
+	Read Memory only for debugger; no I/O, no side effects
+	duplicate read6502 to avoid impacting normal use performances
+*/
+uint8_t
+DEBUGread6502(uint16_t address, uint8_t bank)
```

Comment:
instead of duplicating the existing function, can you create a generic internal function that takes a "bool debug" argument and special cases only the video case, and two frontends, the regular one, and the debug one?

And the same below for the video version.

Reply:
I did it on purpose ;) Here, I thought it would be better to have a specific function for the debugger so when it's not use, the normal flow is totally unaltered vs a bool test on every call. I didn't want to have any impact on the emulator perf. That was the intention.
Now, I agree, it could be a pita to have to maintain 2 functions. Your call, mate.

Reply:
@mist64 So Michael, what do you want me to do ?

Reply:
Branch prediction will limit the cost of the bool check to practically zero. :-)

Reply:
done.